rest/src/main/kotlin/json/request/ThreadRequests.kt

```diff
@@ -0,0 +1,17 @@
+package dev.kord.rest.json.request
+
+import dev.kord.common.entity.optional.Optional
+import dev.kord.common.entity.optional.OptionalBoolean
+import dev.kord.common.entity.optional.OptionalInt
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class ThreadModifyPatchRequest(
+    val name: Optional<String> = Optional.Missing(),
+    val archived: OptionalBoolean = OptionalBoolean.Missing,
+    val autoArchiveDuration: OptionalInt = OptionalInt.Missing,
+    val locked: OptionalBoolean = OptionalBoolean.Missing,
+    @SerialName("rate_limit_per_user")
+    val ratelimitPerUser: OptionalInt = OptionalInt.Missing
+)
```

Comment:
This seems redundant, the request has the same layout as a channel modify request

Reply:
but the channel can't be locker or auto-archived

Reply:
Well yeah, but the modify channel endpoint does accept those fields.

Reply:
you are suggesting we yeet those fields in channel request?

Reply:
Yes, this is how we approach the issue when it comes to other channels and their unique fields as well